vignettes/targets_workflow.Rmd

```diff
@@ -0,0 +1,540 @@
+
+This tutorial shows how to combine the R package `targets`, which orchestrates the components of computationally intensive analysis projects, with `data.validator`, which is a tool for data validation and reporting.
+
+Although the subject of data validation is very broad, we will concentrate on two key ideas:
+
+-   **Metadata validation** with an example of a defensive approach that 'fails fast' and generates a well handled error response. Having a 'fine-grained' error handling, we can perform important actions like send an email, store validation data to a database for analysis, or publish an HTML report generated by `data.validator` to Posit Connect.
+
+-   **Domain expertise knowledge** and **error-localization** that is an essential ingredient to any data curation process. We demonstrate a simple example of a data workflow where `data.validator` is the 'pedal force' of a data cleansing and imputation process.
+
+## File structure
+
+`targets` is an opinionated framework to organize your data workflows. For our purposes, the file structure of the targets project looks like:
+
+    ├── _targets.R
+    ├── data.csv
+    ├── schema.csv
+    ├── R/
+    │   ├── functions.R
+
+-   `data.csv` contains the data that we want to analyze.
+
+-   `schema.csv` contains the schema that will be used to assert the data structure.
+
+-   `R/functions.R` contains our custom user-defined functions.
+
+-   `_targets.R` is the special file that defines the `targets` workflow.
+
+### Input data
+
+Let's consider the `iris` dataset with some tweaks in order to work with different data validation concepts. We perform the following mutations:
+
+1.  Add an index column
+2.  Add a few outliers
+
+and save to a CSV file that will be tracked as a data input by the pipeline.
+
+```{r}
+input_data <- iris %>%
+  mutate(
+    index = 1:nrow(iris),
+    Sepal.Width = if_else(Sepal.Width > 4, 999, Sepal.Width)
+  )
+
+write_csv(input_data, "data.csv")
+```
+
+### Data structure
+
+Let's get the table's schema and save it as a CSV file. This file and the related data can be considered as inputs in the data pipeline. Saving the input table's expected schema to a file can aid with early error detection due to schema changes, is simple to share, and can be versioned. Changes to the schema file can be tracked by the targets pipeline, and any stages that depend on it upstream will be invalidated.
+
+We save the schema as a CSV with two columns. `name` stands for the name of the columns, and `type` stands for their class:
+
+```{r}
+get_schema <- function(data) {
+  imap_dfr(data, \(var, name) {
+    tibble(
+      name = name,
+      type = class(var)
+    )
+  })
+}
+
+get_schema(input_data) %T>%
+  print() %>%
+  write_csv("schema.csv")
+```
+
+## Metadata validation
+
+```{r, include = FALSE, eval = TRUE}
```

Comment:
Please add a comment to all code chunks that are not included with a short description (what is happening and why it is needed).

Reply:
I agree